evm/p2p/test_ecies.py

```diff
@@ -0,0 +1,83 @@
+import pytest
+
+from eth_utils import (
+    decode_hex,
+    encode_hex,
+)
+
+from eth_keys import keys
+
+from evm.p2p import ecies
+
+
+# (pvikey_hex, pubkey_hex, expected_ecdh) tuples with known-good values, to ensure our
+# ECC backends are compatible with other clients'.
+# Copied from
+# https://github.com/ethereum/cpp-ethereum/blob/3c49a0/test/unittests/libp2p/rlpx.cpp#L427
+# and
+# https://github.com/ethereum/go-ethereum/blob/5c9346/crypto/ecies/ecies_test.go#L456
+STATIC_ECDH_VALUES = [
+    ("0x332143e9629eedff7d142d741f896258f5a1bfab54dab2121d3ec5000093d74b",
+     "0xf0d2b97981bd0d415a843b5dfe8ab77a30300daab3658c578f2340308a2da1a07f0821367332598b6aa4e180a41e92f4ebbae3518da847f0b1c0bbfe20bcf4e1",  # noqa: E501
+     "0xee1418607c2fcfb57fda40380e885a707f49000a5dda056d828b7d9bd1f29a08",
+     ),
+    ("0x7ebbc6a8358bc76dd73ebc557056702c8cfc34e5cfcd90eb83af0347575fd2ad",
+     "0x83ede0f19c3c98649265956a4193677b14c338a22de2086a08d84e4446fe37e4e233478259ec90dbeef52f4f6c890f8c38660ec7b61b9d439b8a6d1c323dc025",  # noqa: E501
+     "0x167ccc13ac5e8a26b131c3446030c60fbfac6aa8e31149d0869f93626a4cdf62",
+     ),
+]
+
+
+def test_encrypt_decrypt():
+    msg = b'test yeah'
+    privkey = ecies.generate_privkey()
+    ciphertext = ecies.encrypt(msg, privkey.public_key)
+    decrypted = ecies.decrypt(ciphertext, privkey)
+    assert decrypted == msg
```

Comment:
This is the type of test that I've historically used hypothesis for (things that have a round trip operation like encrypt/decrypt, encode/decode)

Reply:
Oh, I see!